lib/round-tracker.js

```diff
@@ -225,3 +237,27 @@ async function defineTasksForRound (pgClient, sparkRoundNumber) {
     TASKS_PER_ROUND
   ])
 }
+
+/**
+ * @param {Awaited<ReturnType<import('./ie-contract.js').createMeridianContract>>} contract
+ * @param {bigint} roundIndex
+ * @returns {Promise<biging>} Filecoin Epoch (ETH block number) when the SPARK round started
+ */
+export async function getRoundStartEpoch (contract, roundIndex) {
+  assert.strictEqual(typeof roundIndex, 'bigint', `roundIndex must be a bigint, received: ${typeof roundIndex}`)
+
+  // Look at the last 1000 blocks to handle the case when we have an outage and
+  // the rounds are not advanced frequently enough.
+  const recentRoundStartEvents = (await contract.queryFilter('RoundStart', -1000))
+    .map(({ blockNumber, args }) => ({ blockNumber, roundIndex: args[0].toBigInt() }))
```

Comment:
It's unfortunate that the `contract.on(ev, fn)` doesn't expose the block number

Reply:
Agreed 😢